.github/workflows/build-and-publish-container-image.yml

```diff
@@ -1,40 +1,65 @@
 name: Build and Push Container Image
-run-name: "Build and Push Image: Kubectl ${{ inputs.kubectl-version }}, Helm ${{ inputs.helm-version }}, Powershell ${{ inputs.powershell-version }}"
 
 on:
-  workflow_dispatch:
-    inputs:
-      kubectl-version:
-        required: true
-        type: string
-        default: "1.30.5"
-      helm-version:
-        required: true
-        type: string
-        default: "3.16.1"
-      powershell-version:
-        required: true
-        type: string
-        default: "7.4.5"
-      tag-as-latest:
-        description: "(Main only) Tag the image as latest - kubectl 1.30.x only"
-        required: true
-        type: boolean
-        default: false
+  pull_request:
+    paths:
+      - 'versions.json'
+  push:
+    branches:
+      - main
+    paths:
+      - 'versions.json'
+
+env:
+  ArtifactoryImagePath: "${{ secrets.ARTIFACTORY_DOCKER_REPO_HOSTNAME }}/octopusdeploy/kubernetes-agent-tools-base"
+  DockerHubImagePath: "octopusdeploy/kubernetes-agent-tools-base"
 
 jobs:
+  versions:
+    runs-on: ubuntu-latest
+
+    outputs:
+      toolsVersions: ${{ steps.versions.outputs.tools }}
+      latestVersion: ${{ steps.versions.outputs.latest }}
+      revision: ${{ steps.versions.outputs.revision}}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: "Parse versions.json"
+        id: versions
+        run: |
+          toolsVersions=$(jq -c .tools versions.json)
+          latestVersion=$(jq -r -c .latest versions.json)
+          revision=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 6)
```

Comment:
Given that the revision string is now being generated here instead of read from the version.json file, how would Tentacle know which revision to use? i.e. what the image tag will look like for the latest revision

Reply:
Yeah, I've stuffed this 😓 

I'm thinking/working on a solution. Feels like we are over-engineering things here

Reply:
I think the version.json would always need a revision property. What you are thinking of is perhaps automating the process of updating that revision. So whenever a dependency in version.json changes, then an automatic PR is created to update the revision, which once committed will perform the build and push of the images.

Reply:
It might be simpler for now to just start with a manual process to update the revision

Reply:
That's not my concern. My main concern is the revision number not being accurate to the _actual_ revision of that container image